app/livechat/server/api/v1/message.js

```diff
@@ -65,6 +65,45 @@ API.v1.addRoute('livechat/message', {
 });
 
 API.v1.addRoute('livechat/message/:_id', {
+	get() {
+		try {
+			check(this.urlParams, {
+				_id: String,
+			});
+
+			check(this.queryParams, {
+				token: String,
+				rid: String,
+			});
+
+			const { token, rid } = this.queryParams;
+			const { _id } = this.urlParams;
+
+			const guest = findGuest(token);
+			if (!guest) {
+				throw new Meteor.Error('invalid-token');
+			}
+
+			const room = findRoom(token, rid);
+			if (!room) {
+				throw new Meteor.Error('invalid-room');
+			}
+
+			const message = Messages.findOneById(_id);
+			if (!message) {
+				throw new Meteor.Error('invalid-message');
+			}
+
+			const { msg, u, ts, attachments, tmid } = message;
```

Comment:
please instead of getting the whole message object and return only a few fields after destructuring, please get only needed fields from DB, then you can return the whole object.. it will use less network between server <-> db and will do less operations (creating variables and so on).. I'd do something like:
```javascript
const message = Messages.findOneById(_id, { fields: { msg: 1, u: 1, ts: 1, attachments: 1, tmid: 1 } });
if (!message) {
	throw new Meteor.Error('invalid-message');
}

return API.v1.success({ message });
```

btw, why not return all message fields?